extensions/elasticsearch/elasticsearch-6/src/test/java/com/hazelcast/jet/elastic/ElasticSinkBuilderTest.java

```diff
@@ -79,14 +74,11 @@ public void when_writeToFailingSink_then_shouldCloseClient() throws IOException
             // ignore - elastic is not running
         }
 
-        logger.fine("Clients to close: " + ClientHolder.elasticClients.size());
-        for (RestClient client : ClientHolder.elasticClients) {
-            logger.fine("Closing client " + client);
-            verify(client).close();
-        }
-    }
-
-    static class ClientHolder implements Serializable {
-        static Set<RestClient> elasticClients = Collections.synchronizedSet(new HashSet<>());
+        assertTrueEventually(() -> {
```

Comment:
What do you think about moving this assertion to ClientHolder to avoid duplication?
```java
static void assertAllClientsNotRunning() {
        assertTrueEventually(() -> {
            for (RestClient client : elasticClients) {
                CloseableHttpAsyncClient httpClient = getFieldValueReflectively(client, "client");
                assertThat(httpClient.isRunning()).isFalse();
            }
        });
    }
```

Reply:
Good idea 4f579c7271